### ISSUE TITLE
Slow loading after unlock on Android

### ISSUE THREAD

**Issue by novocaiin** [bitwarden/mobile#1723]
### Steps To Reproduce

1. Unlock Bitwarden using fingerprint
2. Try using the UI

### Expected Result

I can start using the app without noticeable delay.

### Actual Result

There is a pretty significant delay before the UI is rendered at all and yet another delay before I can start using it.

### Screenshots or Videos

https://user-images.githubusercontent.com/95007229/150283613-41d53470-44cd-4707-bc0b-5ea632e6b06f.MOV

### Additional Context

This is specific to Android.
I have an iPhone with Touch ID (fingerprint scanner) - one much older and less performant than the Android phone in the video - and it has no noticeable delay between when I touch the scanner and when I can start searching or selecting a login.

This only happens when unlocking using fingerprint is necessary.
As can be seen from the video, if it's unlocked there is no delay.

### Operating System

Android

### Operating System Version

12

### Device

Pixel 5

### Build Version

2.15.0

### Beta

- [ ] Using a pre-release version of the application.

**Comment by iAmSaugata** [comment]
I am having same problem with my Galaxy Note 10 on Android 12

**Comment by gaborbit** [comment]
Same or bit slower here with Samsung galaxy

**Comment by mhamzas** [comment]
Always slow

**Comment by iAmSaugata** [comment]
Just an update, I have changed my handset to Nothing Phone 1, which is much faster phone than my Galaxy Note 10 Lite, but the application still taking huge time to open and load.
I am currently using bitwarden third-party app called Keyguard (https://play.google.com/store/apps/details?id=com.artemchep.keyguard&gl=US), compare to Bitwarden official app, Keyguard opens and load lot more faster. Bitwarden took 17.80 seconds to open and load, where as Keyguard only took only 3.45 seconds.

**Comment by rasved** [comment]
Same problem for me. I typically experience around 7 seconds from the vault is unlocked, till I can use/see the entries. THAT.IS.LOONG.TIME!

I have tried different devices, and different accounts with both 1100+ entries and one with <25. Same problem.

**Comment by InternetSemplice** [comment]
Aslo to me the vault, on the mobile **android** app (Galaxy 40), takes **4-7 secs to be available**. 
It is a lot of time, especially because I login MANY times a day (I am a web professional).
_PLS, try to fix it._

**Comment by i3130002** [comment]
Duplicate of #1567

**Comment by mhamzas** [comment]
> 

Looks like the app is not available on the playstore anymore. Can anybody share the apk ?

**Comment by i3130002** [comment]
> Looks like the app is not available on the playstore anymore. Can anybody share the apk ?

@mhamzas 

It is!!!  (Please do not post unrelated comments to the issue topic)

https://play.google.com/store/apps/details?id=com.x8bit.bitwarden

**Comment by dbosompem** [comment]
Hi all, thank you for reporting this and sharing details. I will pass this on to the team and we will make time to resolve this. Apologies for any inconvenience this may have caused. Thank you for your patience!

**Comment by flaviovs** [comment]
I just want to add that I'm experiencing this issue **even when not using BW with assistive features enabled or auto-filling inside other apps**. Here's what I'm seeing in my Sansumg Galaxy phone with Android 8 and about 450 entries in the vault on a 300Mbps wifi connection:

- Tap BW icon in the Android launcher, wait for the UI to appear and ask for fingerprint: ~11s
- Authenticate using fingerprint, time for the interface to appear and become responsive: ~17s
- Tap the search icon, time to able to type on the search bar: ~10s (note after I tap the search icon it takes ~1s for the icon to show up in "selected" state, it stays like that for ~2s, then normal state, and then the remaining 7s to see the typing cursor)
- Search something: sometimes it is very laggy, sometimes usable
- Tap on an entry: 1-3s to open it
- Scrolling entries sometimes makes the app unresponsive for +10s
- Unlocking the phone or switching apps (e.g. to copy and paste logins or passwords) is usually fast *only if the app is showing a login entry*. If the app is any other screens it usually comes back with a white screen for 2-5s

I hope this is useful for the devs to debug the issue.

**Comment by bond-os** [comment]
I have same issues on Huawei Mate 10 pro with emui 12 - very slow start (+10s), very slow load after unlock (+10s). this seem to be getting worse with time, either with app updates or vault size (400+ entries now).

**Comment by flaviovs** [comment]
@dbosompem any feedback from the dev team? Any ETA for fixing this?

It has been almost 11+ months since this issue was created. It clearly feels that the problem is becoming worst over time, even though the # of entries in my vault is the same. I already had to install another app to manage my TOTP keys because having to wait minutes just to get a code was driving me nuts. The app on Android is barely usable in my phone now. :worried:

**Comment by dbosompem** [comment]
Hi @flaviovs , thanks for checking in. I have passed this on to the team and the team is aware. This has been added to a list of performance improvements we want to look at, for mobile. I will follow up with team so that a higher priority is assigned, and will share with you an ETA once that is certain. Apologies for the inconveniences and thank you for your patience!

**Comment by i3130002** [comment]
Dear @dbosompem, now that you are on it, Please check the firefox extension as it got really bad (Almost unusable)

[Screencast from 12-12-2022 08:19:52 PM.webm](https://user-images.githubusercontent.com/11602376/207104720-269dfda1-77b8-472a-93ff-28d80a86af93.webm)

**Comment by bond-os** [comment]
seems that deleting application data (remove cache + remove data) and logging in again fixes issues on my device - its now fast again. This is first time I did it since logging in on this device aroud 12/2018.

**Comment by SPFZ** [comment]
I tried @bond-os workaround and got a huge speedup! Thank you!
I had about 250MB of Data storage reported before the cleanup and about 6 MB after. I assume something is piling up in the data.
I have the app installed on the same phone for several years as well.

**Comment by bond-os** [comment]
Good point on data size @SPFZ, on my device it also was more than 200MB and now it's around ~6MB.

**Comment by gregmac** [comment]
This worked for me, too. Similar experience: was at ~328MB Data, after syncing 1.3MB.

**Comment by flaviovs** [comment]
@bond-os, thanks for the tip. Granted the app is responsive again after following your suggestion.

@dbosompem, thank you and the team for being on top of this issue.

**Comment by vrosu** [comment]
Recent member (2 months old)
I experience the same issue as the one reported by OP. 
Some details:
-phone is Huawei P30 Pro, non-rooted, 8GB RAM, 256 GB native internal storage, Android 12, Security patch level November 1 2022 
- in my case I experience a 9 seconds delay (spinning circle) between the fingerprint unlock and the application being usable (displaying types, folders, having search box usable). This time does not depend on being on Wifi vs 4G.
- I have ~680 items (355 logins and 321 secure notes) and 5 folders. All secure notes have 4 custom fields and all the logins one custom field (result of an import I did from mSecure). No file attachments.
- I'm a premium user (I'll report this as a bug and link to this comment in there)

@bond-os Unfortunately this did not work for me. Data is 8.4 MB.
@dbosompem fyi, I really hope this can be solved, as I use randomly generated passwords frequently and the simple act of opening Bitwarden on Android is very painful.

**Comment by AChep** [comment]
@vrosu any chance you could test it with [Keyguard](https://www.reddit.com/r/keyguard/comments/104swmg/keyguard_for_bitwarden_goes_open_beta/) _(note that it is a third-party client, not affiliated with Bitwarden)_? Curious about what the performance would be.

**Comment by seanf** [comment]
Seems like this is probably the reason the slowness is so painful: https://github.com/bitwarden/mobile/issues/2006

**Comment by vrosu** [comment]
Commenting to be aware of what worked for me, based on the tech support case I opened.
It seems I had many entries in the trash folder.
In Android, the bitwarden client has the + button exactly on top of the trash folder count... And because of this I never saw that I actually had trash entries.
Also, in the desktop bitwarden client there is no item count... So again I was not aware of the 2600 items in trash.
As soon as I cleaned the trash, the load time went to 3 seconds.
Still slow, but it is not that bothering now

**Comment by i3130002** [comment]
> I'm having this issue too. Clearing local storage hasn't helped. No items in my yeah. I'll considering moving to 1Password or ProtonPass as a result of this.

The loading problem also happens on Browser Extensions but it's only for logging in and it's quite good after that. I wish they could fix this problem as it's disappointing specially while filling the password on the websites.

**Comment by dlecan** [comment]
The Bitwarden Android app gets slower and slower
I subscribed a family plan a year ago, first for me, the for my children. The average opening time is ~5-10s on our android phones (Oppo, Cat).
Almost unuseable
Only my wife is happy: she’s got an iPhone!

We’ll try Keyguard or switch for another password manager.

**Comment by pamperer562580892423** [comment]
I can imagine, this might get better with the new native mobile apps, presumably coming in a few weeks/months… https://www.reddit.com/r/Bitwarden/comments/1b32bbz/going_native_the_future_of_the_bitwarden_mobile/?share_id=8bXQrGpuuXHxb8xiTrlKn&sort=new

**Comment by novocaiin** [comment]
That's lovely news. No amount of reinstalling the app and clearing data has fixed this for me.

**Comment by vvolkgang** [comment]
Issue migrated to https://github.com/bitwarden/mobile/issues/1723
